package org.example.chess;

import org.example.boardgame.Board;
import org.example.boardgame.Position;
import org.example.chess.chess.King;
import org.example.chess.chess.Rook;

public class ChessMatch {

    private Board board;

    public ChessMatch(){
        board = new Board(8,8);
        initialSetup();
    }

    public ChessPiece [][] getPieces(){
        ChessPiece [][] mat = new ChessPiece[board.getRows()] [board.getColumns()];
        for (int i=0; i< board.getRows(); i++){
            for (int j=0; j<board.getColumns(); j++ ){
                mat[i][j] = (ChessPiece) board.piece(i,j);
            }
        }
        return mat;
    }

    private void initialSetup(){
        board.placePiece(new Rook(board, Color.WHITE), new Position(5,1) );
        board.placePiece(new King(board, Color.WHITE),new Position(3,5));
    }


}
